File: app/src/main/java/al/bruno/financaime/util/Converter.java
package al.bruno.financaime.util;

import java.util.Date;

import androidx.room.TypeConverter;
import androidx.room.TypeConverters;

public class Converter {
    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/al/bruno/financaime/util/Converter.kt
package al.bruno.financaime.util

import java.util.Date

import androidx.room.TypeConverter
import androidx.room.TypeConverters

class Converter {
    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }
}
